### ISSUE TITLE
Batch-style PNG-export of heavy ORA files

### ISSUE THREAD

**Issue by snan** [mypaint/mypaint#619]
My computer is eight years old and sometimes it's straining under the weight of all the layers and stuff I want to put in, especially if I generate or manipulate the ORA files programmatically (such as scaling them up).
I'd love, and use probably pretty much daily, a little batch program to convert them to PNG respecting all blend modes, svg layers etc.

**Comment by achadwick** [comment]
We could add a command-line switch like Inkscape's `-e`/`--export-png` to the main MyPiant launcher. Separate scripts are harder because of all the work that would need to be done to replicate the path and do i18n setup.

**Comment by snan** [comment]
Yeah, and from a user perspective, I'd prefer that, actually. I just wasn't sure which would be easier.

**Comment by odysseywestra** [comment]
This can also be a good way to test out a plugin system for mypaint as well.

**Comment by jeisner** [comment]
Since ORA files are zip files that are supposed to contain a mergedimage.png as well as the separate layers, here is my quick-and-dirty bash function to convert valid ORA files to PNG files.
```
function ora2png () { for orafile in "$@"; do unzip -q $orafile mergedimage.png; touch mergedimage.png; mv mergedimage.png "${orafile%.*}.png"; done }
```
I agree that it would be nice for mypaint to have a command-line switch for this.

**Comment by snan** [comment]
Oh, that's something I never discovered. That's interesting and I will get a lot of use from it. Thank you very much.

The downside is that it locks the SVG resolution into the resolution of the current image (i.e. sometimes it can look pretty sweet if you have water color layers in raster masked by svg layers (w/ Destination In), and then use cubic interpolation to scale the water color layers up while their edges are scaled as vector.

**Comment by snan** [comment]
So the problem is the SVG layers. Of many kinds: dst-in, dst-out, src-over, src-multiply and many others.
I want to scale them up as vector and then scale the PNG layers up with cubic interpolation.
Krita can't do it, GIMP can't do it, ImageMagick can't do it.

It would be a smop to make a script to scale up every layer (both png and svg) and adjust the translations in `stack.xml` accordingly. But that obv won't automatically create a `mergedimage.png`. I'd still have to do the actual composition. And again, here maybe some shell scripting with Graphics Magick or ImageMagic or GMIC could solve it. That's a bigger smop but… uh… I guess that's the remaining hope.

**Comment by snan** [comment]
It's not just scaling, and it's not just that my (now 11 year old) machine is slow, it's that I want to do headless post-processing of files. I have scripts that go in the ora files, mess with the layers, and then the only way to get a PNG out of it is to start it up in MyPaint, hit ctrl-S and so forth manually. Kinda a chore when I have dozens of files is what I'm saying.

I get that this is an enhancement and that I should've gotten off my own lazy chair in the last three years, learned to grok mypaint's code and tried to implement this myself. I don't mean to come across as entitled for a rare feature that only I would ever use. But on the other hand it's frustrating that mypaint's take on the ora format is so mypaint-specific. (Not that I want SVG-layers removed, I love them and rely on them.)

**Comment by teleportingtortoise** [comment]
I'm still trying to familiarize myself with the MyPaint code and I definitely have more to learn about coding, but is there any advice on what would need to happen for MyPaint to be able to do export from the command line? Like what parts of the code I should focus on understanding and such? I need this feature enough that I'm willing to fork and take a crack at it even if it doesn't end up turning out or being merge worthy.

**Comment by AesaraB** [comment]
> This can also be a good way to test out a plugin system for mypaint as well.

Related issue #209
